### PR TITLE
Read from IRequest instead of reading twice

### LINKED ISSUE #14541: PUT Routes arguments are not properly parsed
It seems that the AppFramework has a bug.

When I tried to update the e-mail address on a fresh install on the personal page I got errors that I supplied an "Invalid mail address".

After a bit of debugging it seems that $mailAddress on https://github.com/owncloud/core/blob/master/settings/controller/userscontroller.php#L443 is just not set! And thus the address does not validate.

The other routes seem to work fine. But this is the only PUT request.

When tested on the stable8 branch everything works as expected.

### PR DESCRIPTION
**Pull request by LukasReschke**
Potentially fixes https://github.com/owncloud/core/issues/14541 and https://github.com/owncloud/core/issues/14506

… let's see what our Jenkins says to this …

### CODE REVIEW

**Comment by LukasReschke** [comment]
I removed the TemplateLayout tests that I added ages ago since they were horribly anyways and such stuff really prevents from having a sane Request class…

**Comment by LukasReschke** [comment]
Other apps that needs fixing: https://github.com/search?utf8=%E2%9C%93&q=user%3Aowncloud+file_get_contents%28%22php%3A%2F%2Finput%22%29&type=Code&ref=searchresults

*sigh*

**Comment by Raydiation** [comment]
Good job, looks good :+1:

File: lib/private/api.php [hunk 1]
```diff
@@ -84,11 +84,13 @@ public static function register($method, $url, $action, $app,
 	 * @param array $parameters
 	 */
 	public static function call($parameters) {
+		$request = \OC::$server->getRequest();
```
Comment (bantu) [inline lib/private/api.php]: `$method = OC::$server->getRequest()getMethod();` ?

Comment (LukasReschke) [inline lib/private/api.php]: Mhm… Would require me to instantiate another one for `getParams`. This way it's IMHO easier to migrate it in the future to a testable controller as we can just replace `$request` with a member variable.

**Comment by LukasReschke** [comment]
Requires test execution on 5.4 and Co. – as discussed with @DeepDiver1975 yesterday. Will ping him again on monday.

File: lib/private/api.php (hunk 1 above)
Comment (bantu) [inline lib/private/api.php]: Oh err yeah. Could use `$method = $request->getMethod();`.

Comment (LukasReschke) [inline lib/private/api.php]: So true :) :rocket:

Comment (LukasReschke) [inline lib/private/api.php]: Done.

**Comment by LukasReschke** [comment]
:bell: :bell: :bell:  … pinging @DeepDiver1975 again as promised on Friday …  :bell: :bell: :bell:

**Comment by DeepDiver1975** [comment]
@dragotin  @rullzer can I ask you to test this PR with respect to the failing PUT via OCS API? THX

**Comment by rullzer** [comment]
My tests all work as expected again on php 5.5.22! :+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**